lib/galaxy/tools/execute.py

```diff
@@ -48,7 +48,7 @@ def execute_single_job(params):
     burst_at = getattr( config, 'tool_submission_burst_at', 10 )
     burst_threads = getattr( config, 'tool_submission_burst_threads', 1 )
 
-    tool_action = tool.action
+    tool_action = tool.tool_action
```

Comment:
Agreed this is confusing - probably addressing that is beyond the scope of this PR I think.